api/context.md

```diff
@@ -162,13 +162,15 @@ If you want to create the middleware which provides a custom method,
 write like the following:
 
 ```ts
-const app = new Hono()
-
-const echoMiddleware: MiddlewareHandler<{
+interface Env {
   Variables: {
     echo: (str: string) => string
   }
-}> = async (c, next) => {
+}
+
+const app = new Hono<Env>()
+
+const echoMiddleware: MiddlewareHandler<Env> = async (c, next) => {
```

Comment:
I recommend using the `createMiddleware` factory method to avoid defining the types by users them selfs:

```ts
const echoMiddleware = createMiddleware<Env>(async (c, next) => {
  c.set('echo', (str) => str)
  await next()
})
```